website/content/docs/secrets/gcp.mdx

```diff
@@ -211,16 +211,37 @@ below.
 For more information on creating and managing static accounts, see the
 [GCP secrets engine API docs][api] docs.
 
+## Impersonated Accounts
+
+Impersonated accounts are a way to generate a GCP access token that is granted
+the permissions and accesses of another given service account. These access
+tokens do not have the same 10-key limit as service account keys do, yet they
+retain their short-lived nature. By default, their TTL in GCP is 1 hour, but
+this may be configured to be up to 12 hours
+[as per Google Cloud documentation.](https://cloud.google.com/iam/docs/create-short-lived-credentials-delegated#sa-credentials-oauth)
+
+### Examples
+
+To configure a Vault role that impersonates the administrator on the Google
+Cloud project with the cloud platform and compute scopes:
+
+```shell-session
+$ vault write gcp/impersonated-account/my-token-impersonate \
+    service_account_email="projectAdmin@my-project.iam.gserviceaccount.com" \
+    token_scopes="https://www.googleapis.com/auth/cloud-platform,https://www.googleapis.com/auth/compute" \
+    ttl="6h"
+```
+
 ## Usage
 
 After the secrets engine is configured and a user/machine has a Vault token with
-the proper permission, it can generate credentials. Depending on how the roleset
+the proper permission, it can generate credentials. Depending on how the Vault role
 was configured, you can generate OAuth2 tokens or service account keys.
 
 ### Access Tokens
 
-To generate OAuth2 tokens, read from `gcp/token/...`. The roleset must have been
-created as type `access_token`:
+To generate OAuth2 tokens, read from `gcp/.../token`. The role must have been
+created as type `access_token` or must be an impersonated account:
```

Comment:
```suggestion
To generate OAuth2 [access tokens](https://cloud.google.com/docs/authentication/token-types#access),
read from the [`gcp/.../token`](/api-docs/secret/gcp#generate-secret-iam-service-account-creds-oauth2-access-token)
API. The roleset or static account must have been created with a [`secret_type`](/api-docs/secret/gcp#secret_type)
of `access_token` or must be an impersonated account:
```

@MilenaHC @robmonte - I think we can make it clear that all three "role" types support the same access token generation in this section via the suggestion above. That addresses the topic of the discussion thread. Let me know your thoughts.

Reply:
I think that was a great place to explain it for all 3, nice!